Dockerfile

```diff
@@ -1,40 +1,70 @@
-# syntax=docker/dockerfile:1.4
-FROM node:14.21.1-alpine3.15
+FROM node:14.21.1-alpine3.15 AS base
```

Comment:
Can you expand on the changes in this file?

Reply:
You can refer more here https://docs.docker.com/build/building/multi-stage/
On high level you can name a stage and re-use that particular stage

Reply:
I was not asking about that but more about the need for different build types, etc.

Reply:
On high level this is what I am achieving

Updated Dockerfile to support multistage build, have broken build to two stages
- development [Everything is present including dev node modules, tests, and src files along with dist]
 -  Will be used primarily for building and testing purposes
- production Only prod node modules and dist is present
 - Drastically reduces build size
Updated all build spec
- rudderlabs/develop-rudder-transformer Will now contain the larger development builds which can be directly tested also can be run
- rudderlabs/rudder-transformer Will now contain the production image which is much smaller
Even though we are pushing images without tests in production builds but we are testing development counter-parts of the same build before pushing